src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -86,106 +69,233 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
+    const match = sortedNodes.find((el) => el.id[0].toUpperCase() === char);
+    if (match) {
+      cursor.current = visibleNodes.findIndex((el) => el.id === match.id);
+      currentNodeId.current = match.id;
+      focusCurrentNode();
+    }
+  };
+
   const handleKeyDown = (event, item) => {
-    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
-      event.preventDefault();
-      handleOnClick(event, item);
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_SPACE:
+      case KeyCode.KEY_RETURN:
+        event.preventDefault();
+        handleOnClick(event, item);
+        break;
+      case KeyCode.KEY_DOWN:
+        event.preventDefault();
+        handleDownArrow();
+        break;
+      case KeyCode.KEY_UP:
+        event.preventDefault();
+        handleUpArrow();
+        break;
+      case KeyCode.KEY_RIGHT:
+        event.preventDefault();
+        if (currentNodeId.current) {
+          if (document.getElementById(currentNodeId.current).getAttribute('aria-expanded') === 'true') {
+            handleDownArrow();
+          } else if (document.getElementById(currentNodeId.current).getAttribute('aria-haspopup') && (!document.getElementById(currentNodeId.current).getAttribute('aria-expanded') || document.getElementById(currentNodeId.current).getAttribute('aria-expanded') === 'false')) {
+            handleOnClick(event, item);
+          }
+        }
+        break;
+      case KeyCode.KEY_LEFT:
+        event.preventDefault();
+        if (currentNodeId.current) {
+          if (document.getElementById(currentNodeId.current).getAttribute('aria-expanded') === 'true') {
+            handleOnClick(event, item);
+          } else if (!document.getElementById(currentNodeId.current).getAttribute('aria-haspopup') || !document.getElementById(currentNodeId.current).getAttribute('aria-expanded') || document.getElementById(currentNodeId.current).getAttribute('aria-expanded') === 'false') {
+            findParentNode();
+          }
+        }
+        break;
+      case KeyCode.KEY_HOME:
+        event.preventDefault();
+        cursor.current = 0;
+        currentNodeId.current = visibleNodes[cursor.current].id;
+        focusCurrentNode();
+        break;
+      case KeyCode.KEY_END:
+        event.preventDefault();
+        cursor.current = visibleNodes.length - 1;
+        currentNodeId.current = visibleNodes[cursor.current].id;
+        focusCurrentNode();
+        break;
+      default:
+        if (event.nativeEvent.keyCode >= KeyCode.KEY_A && event.nativeEvent.keyCode <= KeyCode.KEY_Z) {
+          event.preventDefault();
+          const char = String.fromCharCode(event.nativeEvent.keyCode);
+          findNodeMatching(char);
+        }
+        break;
     }
   };
 
-  const renderMenuItems = (currentMenuItem, firstLevel) => {
+  const renderMenuItems = (currentMenuItem, parent = '', firstLevel = false) => {
```

Comment:
> I'd recommend setting parent id to `undefined` or an explicit `null` value when at the root of the node tree. 

I left this comment above, I think this should default to something more explicit than empty string